sonoff.server.module.js

```diff
@@ -58,19 +58,33 @@ module.exports.createServer = function (config) {
         }
     }
 
+    function updateDevice(deviceId, index, switchState) {
+        var d = state.getDeviceById(deviceId);
+        if (!d || (typeof d.conn == 'undefined')) return "disconnected";
+
+        var dstate = d.state;
+
+        if(index > dstate.length - 1) return "not found";
+
+        dstate[index]['switch'] = (switchState ? "on" : "off");
+
+        state.pushMessage({ action: 'update', value: {switches : dstate}, target: deviceId });
+    }
+
     state.pushMessage = a => {
         var rq = {
             "apikey": "111111111-1111-1111-1111-111111111111",
+            "selfApikey" : "111111111-1111-1111-1111-111111111111",
             "action": a.action,
             "deviceid": a.target,
             "params": a.value,
             "userAgent": "app",
             "sequence": Date.now().toString(),
-            "ts": 0,
-            "from": "app"
+            // "ts": 0,
```

Comment:
why was this interface changed? are there new informations available about sonoff devices?